src/diagram/Diagram.ts

```diff
@@ -139,3 +140,93 @@ export class Diagram {
     this._fields = m.fields.map(p => new Field(p.name, p.length));
   }
 }
+
+export class Snapshot<T extends Cancellable> {
```

Comment:
Snapshot and Timeline classes should be in a separated file.

Reply:
Two classes has been moved to Timeline.ts